examples/Agents_Function_Calling_Barista_Bot.ipynb

```diff
@@ -118,64 +118,80 @@
+        "    order.append((drink, tuple(modifiers)))\n",
         "\n",
         "\n",
-        "def get_order() -> Iterable[tuple[str, Iterable[str]]]:\n",
-        "  \"\"\"Returns the customer's order.\"\"\"\n",
-        "  return order\n",
+        "def get_order() -> List[Tuple[str, Tuple[str, ...]]]:\n",
+        "    \"\"\"Returns the customer's current order.\"\"\"\n",
+        "    return order.copy()\n",
         "\n",
         "\n",
         "def remove_item(n: int) -> str:\n",
-        "  \"\"\"Remove the nth (one-based) item from the order.\n",
+        "    \"\"\"Removes the nth (one-based) item from the order.\n",
         "\n",
-        "  Returns:\n",
-        "    The item that was removed.\n",
-        "  \"\"\"\n",
-        "  item, modifiers = order.pop(int(n) - 1)\n",
-        "  return item\n",
+        "    Args:\n",
+        "        n (int): The one-based index of the item to remove.\n",
+        "\n",
+        "    Returns:\n",
+        "        str: The name of the removed item.\n",
+        "    \n",
+        "    Raises:\n",
+        "        ValueError: If the provided index is out of range.\n",
+        "    \"\"\"\n",
+        "    if 1 <= n <= len(order):\n",
+        "        item, _ = order.pop(n - 1)\n",
+        "        return item\n",
+        "    else:\n",
+        "        raise ValueError(\"Invalid item number. Please provide a valid index.\")\n",
         "\n",
         "\n",
         "def clear_order() -> None:\n",
-        "  \"\"\"Removes all items from the customer's order.\"\"\"\n",
-        "  order.clear()\n",
+        "    \"\"\"Removes all items from the customer's order.\"\"\"\n",
+        "    order.clear()\n",
         "\n",
         "\n",
         "def confirm_order() -> str:\n",
-        "  \"\"\"Asks the customer if the order is correct.\n",
-        "\n",
-        "  Returns:\n",
-        "    The user's free-text response.\n",
-        "  \"\"\"\n",
-        "\n",
-        "  print('Your order:')\n",
-        "  if not order:\n",
-        "    print('  (no items)')\n",
+        "    \"\"\"Asks the customer if the order is correct.\n",
         "\n",
-        "  for drink, modifiers in order:\n",
-        "    print(f'  {drink}')\n",
-        "    if modifiers:\n",
-        "      print(f'   - {\", \".join(modifiers)}')\n",
+        "    Returns:\n",
+        "        str: The user's response.\n",
+        "    \"\"\"\n",
+        "    print('Your order:')\n",
+        "    if not order:\n",
+        "        print('  (no items)')\n",
+        "    else:\n",
+        "        for drink, modifiers in order:\n",
+        "            print(f'  {drink}')\n",
+        "            if modifiers:\n",
+        "                print(f'   - {\", \".join(modifiers)}')\n",
         "\n",
-        "  return input('Is this correct? ')\n",
+        "    return input('Is this correct? (yes/no): ').strip().lower()\n",
```

Comment:
This isn't necessary - part of the "magic" here is that the user can say whatever they want and the model will interpret it.